src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public partial class DependencyContextJsonReader : IDependencyContextReader
+    {
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            DependencyContext dependencyContext;
+            try
+            {
+                dependencyContext = ReadCore(drained.AsSpan(0, drained.Count));
```

Comment:
Won't the parameterless version of AsSpan() use the array segment's offset and count implicitly?  That seems better, not only because it's simpler, but also because this current code is assuming that the returned array segment begins at offset 0, which if ReadToEnd changed, it might not.